app/scripts/views/complete_account_unlock.js

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+define([
+  'views/form',
+  'views/base',
+  'stache!templates/complete_account_unlock',
+  'lib/auth-errors',
+  'lib/validate'
+],
+function (FormView, BaseView, CompleteAccountUnlockTemplate, AuthErrors,
+    Validate) {
+
+  function doesLinkValidate(uid, code, isLinkDamaged) {
+    return Validate.isUidValid(uid) &&
+           Validate.isCodeValid(code) &&
+           ! isLinkDamaged;
+  }
+
+  var CompleteAccountUnlockView = FormView.extend({
+    template: CompleteAccountUnlockTemplate,
+    className: 'complete_account_unlock',
+
+    beforeRender: function () {
+      try {
+        this.importSearchParam('uid');
+        this.importSearchParam('code');
+      } catch(e) {
+        this.logScreenEvent('link_damaged');
+        // This is an invalid link. Abort and show an error message
+        // before doing any more checks.
+        return true;
+      }
+
+      // Remove any spaces that are probably due to a MUA adding
+      // line breaks in the middle of the link.
+      this.uid = this.uid.replace(/ /g, '');
+      this.code = this.code.replace(/ /g, '');
+
+      if (! doesLinkValidate(this.uid, this.code, this._isLinkDamaged)) {
```

Comment:
I was a bit lazy here because I hope https://github.com/mozilla/fxa-content-server/pull/2125 lands first and I can factor all of this out into a standalone model.